### PR TITLE
Fabo/fix extension detection

### PR DESCRIPTION
**Pull request by faboweb**
Signals availability after the listeners are setup

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5f58e6b785492d3760fbaf27--lunieio.netlify.app

File: app/src/ActionModal/components/UndelegationModal.vue [hunk 1]
```diff
@@ -462,13 +459,12 @@ export default {
...
           networkId: this.network,
-          address: this.address,
-          denom: this.stakingDenom,
+          address: this.address
         }
       },
       /* istanbul ignore next */
       update(data) {
-        return data.balance || { amount: 0 }
+        return data.balancesV2.find(({denom}) => this.stakingDenom) || { total: 0, available: 0 }
```
Comment (Bitcoinera) [inline app/src/ActionModal/components/UndelegationModal.vue:467]: What are these changes? :eyes:

Comment (faboweb) [reply app/src/ActionModal/components/UndelegationModal.vue:467]: ups. will remove, when you tested. that was from fixing the undelegation modal stake displaying

Comment (Bitcoinera) [reply app/src/ActionModal/components/UndelegationModal.vue:467]: OK, so I should test this PR as it is right now? To confirm your hypothesis I guess

Comment (faboweb) [reply app/src/ActionModal/components/UndelegationModal.vue:467]: you can, but if the extension works for you know, we can just close it

Comment (Bitcoinera) [reply app/src/ActionModal/components/UndelegationModal.vue:467]: Today it is not working again. Let me try this branch

Comment (Bitcoinera) [reply app/src/ActionModal/components/UndelegationModal.vue:467]: with this branch I don't get to see any address at all:

image

Comment (Bitcoinera) [reply app/src/ActionModal/components/UndelegationModal.vue]: oh I see. I need to rebuild extension...

Comment (Bitcoinera) [reply app/src/ActionModal/components/UndelegationModal.vue]: yes, it works!! :tada: 

image

**Comment by Bitcoinera** [comment]
Let's get this merged! :woman_dancing:

**Comment by Bitcoinera** [comment]
I also tried signing. It all works perfect

File: extension/src/contentScript.js [hunk 2]
```diff
@@ -77,23 +77,24 @@ let listener = (event) => {
...
     listenToWebsiteMessages()
+
+    // we signal that the extension is available so Lunie knows it can start sending requests
+    signalExtensionAvailable()
+  } else {
+    // if there was no request from Lunie yet, we wait until there is
+    setTimeout(checkIfRequestReceived, 100)
   }
 }
 window.onload = () => {
```
Comment (Bitcoinera) [inline extension/src/contentScript.js:97]: when is this triggered? When opening the extension itself?

File: extension/src/contentScript.js [hunk 3]
```diff
@@ -77,23 +77,24 @@ let listener = (event) => {
...
+  if (initExtensionMessageReceived) {
+    // removing listener
+    window.removeEventListener('message', listener, false)
+
     // initialize default listeners
     listenToExtensionMessages()
     listenToWebsiteMessages()
+
+    // we signal that the extension is available so Lunie knows it can start sending requests
+    signalExtensionAvailable()
```
Comment (Bitcoinera) [inline extension/src/contentScript.js:91]: nice

File: extension/src/contentScript.js (hunk 2 above)
Comment (faboweb) [reply extension/src/contentScript.js:97]: no when the page the script is injected in, loaded

File: extension/src/contentScript.js [hunk 4]
```diff
@@ -77,23 +77,24 @@ let listener = (event) => {
...
+  } else {
+    // if there was no request from Lunie yet, we wait until there is
+    setTimeout(checkIfRequestReceived, 100)
```
[the remark this answers is not in the document]
Comment (faboweb) [reply extension/src/contentScript.js:94]: it is. it is calling the outer function

[the remark this answers is not in the document]
Comment (Bitcoinera) [reply extension/src/contentScript.js:94]: ah ok. I see

**Review by Bitcoinera — APPROVED** [review]

**Pull request merged**